gradio/ranged_response.py

```diff
@@ -0,0 +1,185 @@
+# Taken from https://gist.github.com/kevinastone/a6a62db57577b3f24e8a6865ed311463
+# Context: https://github.com/encode/starlette/pull/1090
+
+import os
+import re
+import stat
+import typing as t
+from urllib.parse import quote
+
+import aiofiles
+from aiofiles.os import stat as aio_stat
+from starlette.datastructures import Headers
+from starlette.exceptions import HTTPException
+from starlette.responses import Response, guess_type
+from starlette.staticfiles import StaticFiles
+from starlette.types import Receive, Scope, Send
+
+RANGE_REGEX = re.compile(r"^bytes=(?P<start>\d+)-(?P<end>\d*)$")
+
+
+PathLike = t.Union[str, "os.PathLike[str]"]
```

Comment:
You don't need quotes around types anymore I believe

Reply:
see above, this is third-party code

Reply:
nvm above comment, fixed